controllers/controller-mpd.js

```diff
@@ -65,254 +72,303 @@ function ControllerMpd(nHost, nPort, commandRouter) {
-  self.commandRouter.pushConsoleMessage('[' + Date.now() + '] ' + 'ControllerMpd::pause');
+	var self = this;
+	self.commandRouter.pushConsoleMessage('[' + Date.now() + '] ' + 'ControllerMpd::pause');
 
-  return self.sendMpdCommand('pause', []);
+	return self.sendMpdCommand('pause', []);
 };
 
 // MPD resume
 ControllerMpd.prototype.resume = function() {
-  var self = this;
-  self.commandRouter.pushConsoleMessage('[' + Date.now() + '] ' + 'ControllerMpd::resume');
+	var self = this;
+	self.commandRouter.pushConsoleMessage('[' + Date.now() + '] ' + 'ControllerMpd::resume');
 
-  return self.sendMpdCommand('play', []);
+	return self.sendMpdCommand('play', []);
 };
 
 // MPD music library
 ControllerMpd.prototype.getTracklist = function() {
-  var self = this;
-  self.commandRouter.pushConsoleMessage('[' + Date.now() + '] ' + 'ControllerMpd::getTracklist');
-
-  return self.tracklistReady
-  .then(function() {
-    return self.tracklist;
-  });
+	var self = this;
+	self.commandRouter.pushConsoleMessage('[' + Date.now() + '] ' + 'ControllerMpd::getTracklist');
+
+	return self.mpdReady
+		.then(function() {
+			return libQ.nfcall(libFast.bind(self.clientMpd.sendCommand, self.clientMpd), libMpd.cmd('listallinfo', []));
+		})
+		.then(libFast.bind(self.parseListAllInfoResult, self))
+		.then(function(objResult) {
+			return objResult.tracks;
+		});
 };
 
 // Internal methods ---------------------------------------------------------------------------
 // These are 'this' aware, and may or may not return a promise
 
+// Parses the info out of the 'listallinfo' MPD command
+// Metadata fields to roughly conform to Ogg Vorbis standards (http://xiph.org/vorbis/doc/v-comment.html)
+ControllerMpd.prototype.parseListAllInfoResult = function(sInput) {
+	var arrayLines = sInput.split('\n');
+	var objReturn = {};
+	var curEntry = {}
+
+	objReturn.tracks = [];
+	objReturn.playlists = [];
+
+	for (var i = 0; i < arrayLines.length; i++) {
+		var arrayLineParts = libFast.map(arrayLines[i].split(':'), function (sPart) {
+			return sPart.trim();
+		});
+
+		if (arrayLineParts[0] === 'file') {
+			var objNewMetadata = {'title': '', 'artists': [], 'album': '', 'genres': []};
+			curEntry = {'service': 'mpd', 'uri': arrayLineParts[1], 'metadata': objNewMetadata};
+			objReturn.tracks.push(curEntry);
+		} else if (arrayLineParts[0] === 'playlist') {
+			var objNewMetadata = {'title': '', 'artists': [], 'album': '', 'genres': []};
+			curEntry = {'service': 'mpd', 'uri': arrayLineParts[1], 'metadata': objNewMetadata};
+			objReturn.playlists.push(curEntry);
+		} else if (arrayLineParts[0] === 'Time') {
+			curEntry.metadata.duration = arrayLineParts[1];
+		} else if (arrayLineParts[0] === 'Title') {
+			curEntry.metadata.title = arrayLineParts[1];
+		} else if (arrayLineParts[0] === 'Artist') {
+			curEntry.metadata.artists = libFast.map(arrayLineParts[1].split(','), function (sArtist) {
+				// TODO - parse other options in artist string, such as "feat."
```

Comment:
Good idea! What do you think about storing the MPD "AlbumArtist" as "performer" on Volumio? That would conform to the Ogg Vorbis standard that I've been using so far.